lib/rules/no-uninstalled-addons.ts

```diff
@@ -228,33 +229,55 @@ export = createStorybookRule({
       }
     }
 
+    function findAddonsPropAndReport(node: TSESTree.ObjectExpression) {
+      const addonsProp = node.properties.find(
+        (prop): prop is TSESTree.Property =>
+          isProperty(prop) && isIdentifier(prop.key) && prop.key.name === 'addons'
+      )
+
+      if (addonsProp?.value && isArrayExpression(addonsProp.value)) {
+        reportUninstalledAddons(addonsProp.value)
+      }
+    }
+
     //----------------------------------------------------------------------
     // Public
     //----------------------------------------------------------------------
 
     return {
       AssignmentExpression: function (node) {
         if (isObjectExpression(node.right)) {
-          const addonsProp = node.right.properties.find(
-            (prop): prop is TSESTree.Property =>
-              isProperty(prop) && isIdentifier(prop.key) && prop.key.name === 'addons'
-          )
-
-          if (addonsProp && addonsProp.value && isArrayExpression(addonsProp.value)) {
-            reportUninstalledAddons(addonsProp.value)
-          }
+          findAddonsPropAndReport(node.right)
         }
       },
       ExportDefaultDeclaration: function (node) {
+        if (isIdentifier(node.declaration)) {
```

Comment:
We have a `getMetaObjectExpression` utility function that analyzes a default export node and gives a valid meta object, and that supports all the notations, alongside their TS variants:
- default export {}
- const meta; export default meta;

I'm so sorry that you spent time in this without knowing it! But could you replace the implementation with something like this (and keep the tests):

```ts
import { getMetaObjectExpression } from '../utils'
// ...
      ExportDefaultDeclaration: function (node) {
        const meta = getMetaObjectExpression(node, context)
        if (!meta) {
          return null
        }

        const addonsProp = meta.properties.find(
          (prop): prop is TSESTree.Property =>
            isProperty(prop) && isIdentifier(prop.key) && prop.key.name === 'addons'
        )

        if (addonsProp && addonsProp.value && isArrayExpression(addonsProp.value)) {
          reportUninstalledAddons(addonsProp.value)
        }
      }
```

Reply:
thanks @yannbf! No worries at all, I'm not too familiar with ASTs and eslint plugins - thats a nice util 💯

Reply:
bump @yannbf 🙏